tests/PreciousTest.php

```diff
@@ -233,4 +233,20 @@ public function testIsset()
         $this->assertTrue(isset($f->a), '$f->a is set');
         $this->assertTrue(isset($f->b), '$f->b is set');
     }
+
+    public function testWith()
+    {
+        $f = new F(['a' => 42, 'b' => 3.15]);
+        $f2 = $f->with(['b' => 5.34]);
+
```

Comment:
I would add this check:
```suggestion

        $this->assertNotSame($f, $f2);
```

Reply:
Good point! I'll include that.